flask_static_digest/__init__.py

```diff
@@ -55,10 +54,19 @@ def _load_manifest(self, endpoint, scaffold):
         except (FileNotFoundError, Exception):
             pass
 
-    def _prepend_host_url(self, host, filename):
-        return urljoin(
-            self.host_url, "/".join([self.static_url_path, filename])
-        )
+    def _custom_url_join(self, url):
+        """
+        This function will join the host URL with another URL.
+        It differs from urljoin in that it will keep the path of the host URL
+        even if url is an absolute path.
+        :param url: the URL returned by url_for
+        :type url: str
+        :return: Joined URL
+        """
+        if self.host_url is None:
+            return url
+
+        return self.host_url.rstrip("/") + '/' + url.lstrip("/")
```

Comment:
Rather than concatenating the string with `+` what do you think about `"/".join([self.host_url.rstrip("/"), url.lstrip("/")])`?

Reply:
Either work. Your suggestion is probably more "modern" for lack of a better term. I opted for the `+` way of concatenating because it looked more intuitive and obvious what's returned to me. Another alternative would be using an f-string `f'{self.host_url.rstrip("/")}/{url.lstrip("/")}'`. There's also the option to use the `removesuffix("/")` and `removeprefix("/")` instead of `rstrip("/")` and `lstrip("/")` and while I think those methods would be more fitting, they're relatively new and require Python 3.9 so I opted for the older version to achieve a better compatibility.
In any case, it's your repository so while I respect that you formulated it as a question, I'll ultimately refer to your choice.

Reply:
Yep, I'm dumb. Let's go with an f-string. Given Python 3.9 isn't end of life until October 2025 let's avoid the new functions for now.